wavefront_sdk/direct.py

```diff
@@ -228,24 +236,62 @@ def _batch_report(self, batch_line_data, data_format, entity_prefix,
             # TODO: Please replace a generic Exception with a specific one.
             except Exception as error:
                 logging.error(
-                    'Failed to report %s data points to wavefront %s',
+                    'Failed to report %s data points to wavefront %s\n'
+                    'Data will be requeued and resent.',
                     data_format, error)
+                self._requeue(batch, data_format, data_buffer,
```

Comment:
Let's drop the points if it's a 401 or 403 since that indicates the api token is wrong or the entity type is disabled, otherwise the buffer will get filled up. See Java SDK for example, although no need to "discard until the service is restarted" like the Java SDK does.